Gems/PhysX/Code/Source/EditorShapeColliderComponent.cpp

```diff
@@ -849,4 +939,46 @@ namespace PhysX
     {
         return m_colliderConfig.m_isTrigger;
     }
+
+    void EditorShapeColliderComponent::UpdateTriggerSettings()
+    {
+        if (m_shapeType == ShapeType::QuadSingleSided || m_shapeType == ShapeType::QuadDoubleSided)
+        {
+            if (!m_previousIsTrigger.has_value())
+            {
+                m_previousIsTrigger = m_colliderConfig.m_isTrigger;
+                m_colliderConfig.m_isTrigger = false;
+            }
+            m_colliderConfig.SetPropertyVisibility(Physics::ColliderConfiguration::PropertyVisibility::IsTrigger, false);
+        }
+        else
+        {
+            if (m_previousIsTrigger.has_value())
+            {
+                m_colliderConfig.m_isTrigger = m_previousIsTrigger.value();
+                m_previousIsTrigger.reset();
+            }
+            m_colliderConfig.SetPropertyVisibility(Physics::ColliderConfiguration::PropertyVisibility::IsTrigger, true);
+        }
+    }
+
+    void EditorShapeColliderComponent::UpdateSingleSidedSettings()
+    {
+        if (GetEntity()->FindComponent<EditorRigidBodyComponent>())
+        {
+            if (!m_previousSingleSided.has_value())
+            {
+                m_previousSingleSided = m_singleSided;
+            }
+            m_singleSided = false;
+        }
+        else
+        {
+            if (m_previousSingleSided.has_value())
+            {
+                m_singleSided = m_previousSingleSided.value();
+                m_previousSingleSided.reset();
+            }
```

Comment:
suggestion: There's a pretty nice API called `std::exchange` (https://en.cppreference.com/w/cpp/utility/exchange) that you could potentially use here instead, something like...

```
m_singleSided = std::exchange(m_previousSingleSided, {});
```

Reply:
I'd forgotten about that one. But for me that actually adds more mental overhead versus one extra line that is very simple to parse.